x-pack/plugins/fleet/public/applications/fleet/components/link_and_revision.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiLink, EuiText } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import React, { CSSProperties, memo } from 'react';
+import { EuiLinkProps } from '@elastic/eui/src/components/link/link';
+
+const MIN_WIDTH: CSSProperties = { minWidth: 0 };
+const NO_WRAP_WHITE_SPACE: CSSProperties = { whiteSpace: 'nowrap' };
+
+export type LinkAndRevisionProps = EuiLinkProps & {
+  revision?: string | number;
+};
+
+/**
+ * Components shows a link for a given value along with a revision number to its right. The display
+ * value is truncated if it is longer than the width of where it is displayed, while the revision
+ * always remain visible
+ */
+export const LinkAndRevision = memo<LinkAndRevisionProps>(
```

Comment:
FYI - I would like to add to this component a `isOutdated: boolean` prop and then refactor the Agents List view to also use this component. I have a feeling that the outdated indicator will also be wanted in other scenarios.

Reply:
this would also be good for our Endpoint list

Reply:
yes @kevinlog , we would likely have a similar component in our code base.